src/main/java/net/smartcosmos/dao/relationships/impl/RelationshipPersistenceService.java

```diff
@@ -114,31 +154,48 @@ public RelationshipPersistenceService(
         String targetUrn,
         String relationshipType) {
 
-        UUID accountId = UuidUtil.getUuidFromUrn(tenantUrn);
-        Optional<RelationshipEntity> entity = Optional.empty();
-
         try {
-            entity = relationshipRepository.findByTenantIdAndSourceTypeAndSourceIdAndRelationshipTypeAndTargetTypeAndTargetId(
-                accountId,
-                sourceType,
-                UuidUtil.getUuidFromUrn(sourceUrn),
-                relationshipType,
-                targetType,
-                UuidUtil.getUuidFromUrn(targetUrn));
-        } catch (IllegalArgumentException e) {
-            // Optional.empty() will be returned anyway
-            exceptionLogger("RelationshipPersistenceService:findSpecific", e, sourceType, sourceUrn, relationshipType, targetType);
-        }
+            UUID accountId = UuidUtil.getUuidFromUrn(tenantUrn);
+            Optional<RelationshipEntity> entity = relationshipRepository
+                .findByTenantIdAndSourceTypeAndSourceIdAndRelationshipTypeAndTargetTypeAndTargetId(
+                    accountId,
+                    sourceType,
+                    UuidUtil.getUuidFromUrn(sourceUrn),
+                    relationshipType,
+                    targetType,
+                    UuidUtil.getUuidFromUrn(targetUrn));
+
+            if (entity.isPresent()) {
+                final RelationshipResponse response = conversionService.convert(entity.get(), RelationshipResponse.class);
```

Comment:
no need for the local instance, `return Optional.ofNullable(conversionService.convert(entity.get())`